drivers/modem/Kconfig.wncm14a2a

```diff
@@ -11,6 +11,7 @@ menuconfig MODEM_WNCM14A2A
 	depends on MODEM
 	select MODEM_RECEIVER
 	select NET_OFFLOAD
+	select GPIO
```

Comment:
Can't we clean up few of the options below now?

Reply:
Cleaned up the HW specific options.